File: arrays/container_with_most_water/ts/solution_bruteforce.ts
export const maxAreaBruteforce = (height: number[]): number => {
	let res = 0

	for (let i = 0; i < height.length - 1; i++) {
		for (let j = height.length - 1; j > i; j--) {
			const length = j - i

			if (length * height[i] < res) {
				break
			}

			let minHeight = height[i]

			if (height[j] < minHeight) {
				minHeight = height[j]
			}

			const square = minHeight * length

			if (square > res) {
				res = square
			}
		}
	}

	return res
}

File: _posts/arrays/container_with_most_water/go/solution_bruteforce.go
package main

func maxAreaBruteforce(height []int) int {
	res := 0
	for i := 0; i < len(height)-1; i++ {
		for j := len(height) - 1; j > i; j-- {
			length := j - i
			if length*height[i] < res {
				break
			}

			minHeight := height[i]
			if height[j] < minHeight {
				minHeight = height[j]
			}

			square := minHeight * length
			if square > res {
				res = square
			}
		}
	}

	return res
}
